### PR TITLE
Update connection stats with sync_fetch_and_add

### PR DESCRIPTION
**Pull request by leeavital**
Use __sync_fetch_and_add to change connection stats values as described in https://prototype-kernel.readthedocs.io/en/latest/bpf/ebpf_maps.html#kernel-side-ebpf-program 

This should avoid race conditions where two processors are handling messages for the same connection stat and try to increment the send/received bytes count.

### CODE REVIEW

File: ebpf/c/tracer-ebpf.c [hunk 1]
```diff
@@ -451,19 +451,16 @@ static void update_conn_stats(
     t.sport = ntohs(t.sport); // Making ports human-readable
     t.dport = ntohs(t.dport);
 
+    // initialize-if-no-exist the connection stat, and load it
+    conn_stats_ts_t empty = {};
+    bpf_map_update_elem(&conn_stats, &t, &empty, BPF_NOEXIST);
     val = bpf_map_lookup_elem(&conn_stats, &t);
+
     // If already in our map, increment size in-place
     if (val != NULL) {
```
Comment (leeavital) [inline ebpf/c/tracer-ebpf.c]: technically the val != NULL check is always going to be true, but the BPF verifier will reject this code if we don't check

File: ebpf/c/tracer-ebpf.c [hunk 2]
```diff
@@ -451,19 +451,16 @@ static void update_conn_stats(
     t.sport = ntohs(t.sport); // Making ports human-readable
     t.dport = ntohs(t.dport);
 
+    // initialize-if-no-exist the connection stat, and load it
+    conn_stats_ts_t empty = {};
+    bpf_map_update_elem(&conn_stats, &t, &empty, BPF_NOEXIST);
```
Comment (sunhay) [inline ebpf/c/tracer-ebpf.c]: Isn't this less efficient for cases we're we are just incrementing (which should be the majority of cases where this is called)? We'll be doing two lookups.

Comment (leeavital) [reply ebpf/c/tracer-ebpf.c]: Yes, but I don't know of a better way to do this.

Comment (sunhay) [reply ebpf/c/tracer-ebpf.c]: Whats wrong with the previous method? Were we hitting race conditions?

Comment (leeavital) [reply ebpf/c/tracer-ebpf.c]: There's a race condition with the following sequence:

- thread 1 handles a tcp_send event (a), gets an empty entry
- thread 2 handles a tcp_send event (b), gets an empty entry
 - thread one decides to pupulate the map for the first time, with sent = a.sent
- thread two decides to populate the map for the first time with sent = b.sent


Really we want the entry to be populated with a.sent + b.sent

Comment (sunhay) [reply ebpf/c/tracer-ebpf.c]: Good to know - were you able to reproduce that bug in tests?

Comment (leeavital) [reply ebpf/c/tracer-ebpf.c]: I was not. but nothing in the docs indicate that map access is safe across cores so I think this is worth adding.

**Pull request merged**